src/Ldap.php

```diff
@@ -133,7 +133,11 @@ public function getResource()
     public function getLastErrorCode()
     {
         ErrorHandler::start(E_WARNING);
-        $ret = ldap_get_option($this->resource, LDAP_OPT_ERROR_NUMBER, $err);
+        if (is_resource($this->resource)) {
+            $ret = ldap_get_option($this->resource, LDAP_OPT_ERROR_NUMBER, $err);
+        } else {
+            $ret = false;
+        }
```

Comment:
```suggestion
        $ret = false;
        if (is_resource($this->resource)) {
            $ret = ldap_get_option($this->resource, LDAP_OPT_ERROR_NUMBER, $err);
        }
```

We can also preset the `$ret` variable here with `false` and eliminate the `else`

Reply:
Done